src/solvers/dgsem_tree/container_viscous_2d.jl

```diff
@@ -0,0 +1,61 @@
+mutable struct ViscousContainer2D{uEltype <: Real}
+    u_transformed::Array{uEltype, 4}
+    gradients::Vector{Array{uEltype, 4}}
+    flux_viscous::Vector{Array{uEltype, 4}}
```

Comment:
Sounds good. Remember you can always just infer the type via `ViscousContainer2D{Tgradients, Tflux_viscous, uEltype <: Real}` and `gradients::Tgradients`, `flux_viscous::Tflux_viscous`

Reply:
I just tried `Tuple` instead of `Vector` - this is in principle possible but leads to longer code and is, from my point of view, a bit confusing given the `Vector` datastructures for the internal, resizable structures.

Reply:
To give an example why would need longer code: For `gradients` as a `Tuple` this 

```julia
viscous_container.gradients[dim] = unsafe_wrap(Array,
                                                       pointer(viscous_container._gradients[dim]),
                                                       (nvariables(equations),
                                                        nnodes(dg), nnodes(dg),
                                                        nelements(dg, cache)))
```                                                        
does not work, one would need to take a detour like

```julia
gradients_x, gradients_y = gradients
gradients_x = unsafe_wrap(Array,
                                                       pointer(viscous_container._gradients[1]),
                                                       (nvariables(equations),
                                                        nnodes(dg), nnodes(dg),
                                                        nelements(dg, cache)))
                                                        
gradients_y = unsafe_wrap(Array,
                                                       pointer(viscous_container._gradients[2]),
                                                       (nvariables(equations),
                                                        nnodes(dg), nnodes(dg),
                                                        nelements(dg, cache)))                           
                                                        
gradients = (gradients_x, gradients_y)                                                                                     
```

Reply:
That makes perfect sense now, thanks. The immutability prevents one from changing the pointer to the array...

Reply:
Does it have an issue with SVector{2}?